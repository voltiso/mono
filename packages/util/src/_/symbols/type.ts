// â €â“¥ 2025     ðŸŒ©    ðŸŒ©     â €   â €
// â €         ðŸŒ© VÍ›oÍ›Í›Í›ltÍ›Í›Í›iÍ›Í›Í›Í›soÍ›Í›Í›.comâ €  â €â €â €

/* eslint-disable @typescript-eslint/no-unnecessary-condition */
/* eslint-disable es-x/no-global-this */

declare global {
	namespace Voltiso {
		const TYPE: unique symbol
		type TYPE = typeof TYPE
	}
}
globalThis.Voltiso ??= {} as never
;(Voltiso.TYPE as any) ??= /* @__PURE__ */ Symbol.for('@voltiso/util/TYPE')
export type TYPE = Voltiso.TYPE
export const TYPE: TYPE = /* @__PURE__ */ Voltiso.TYPE
