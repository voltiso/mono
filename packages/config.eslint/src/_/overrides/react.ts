// â €â“¥ 2022     ðŸŒ©    ðŸŒ©     â €   â €
// â €         ðŸŒ© VÍ›oÍ›Í›Í›ltÍ›Í›Í›iÍ›Í›Í›Í›soÍ›Í›Í›.comâ €  â €â €â €

import { defineEslintConfigOverride } from '@voltiso/config.eslint.lib'

// import { codeFiles } from '../files.js'

export const react = defineEslintConfigOverride({
	files: '*',
	// files: codeFiles,

	plugins: ['react', 'react-hooks', 'react-native'],

	extends: [
		// React - CRA
		'react-app',

		'plugin:react/recommended',
		'plugin:react/jsx-runtime',
		'plugin:react-hooks/recommended',
		'plugin:react-native/all',
	],
})
