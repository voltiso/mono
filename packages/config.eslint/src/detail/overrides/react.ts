// â €â“¥ 2022     ðŸŒ©    ðŸŒ©     â €   â €
// â €         ðŸŒ© VÍ›oÍ›Í›Í›ltÍ›Í›Í›iÍ›Í›Í›Í›soÍ›Í›Í›.comâ €  â €â €â €

import { defineEslintConfigOverride } from '@voltiso/config.eslint.lib'

import { codeFiles } from '~/detail/files'

export const react = defineEslintConfigOverride({
	files: codeFiles,

	plugins: ['react'],

	rules: {
		'react/boolean-prop-naming': 1,
		'react/button-has-type': 1,
		'react/default-props-match-prop-types': 1,

		'react/destructuring-assignment': 0, // god, why...

		'react/display-name': 1,
		'react/forbid-component-props': 1,
		'react/forbid-dom-props': 1,
		'react/forbid-elements': 1,
		'react/forbid-foreign-prop-types': 1,
		'react/forbid-prop-types': 1,

		'react/function-component-definition': [
			'warn',
			{
				namedComponents: 'arrow-function',
				unnamedComponents: 'arrow-function',
			},
		],

		'react/hook-use-state': 1,
		'react/iframe-missing-sandbox': 1,
		'react/jsx-boolean-value': 1,
		'react/jsx-child-element-spacing': 1,
		'react/jsx-closing-bracket-location': 1,
		'react/jsx-closing-tag-location': 1,
		'react/jsx-curly-spacing': 1,
		'react/jsx-curly-newline': 1,
		'react/jsx-equals-spacing': 1,

		'react/jsx-filename-extension': [
			'warn',
			{ allow: 'as-needed', extensions: ['.jsx', '.tsx'] },
		],

		'react/jsx-first-prop-new-line': 1,

		'react/jsx-handler-names': 0, // bugged - does not allow `handlers.nameChanged`

		// 'react/jsx-handler-names': [
		// 	'warn',
		// 	{
		// 		eventHandlerPrefix: '',
		// 	},
		// ],

		'react/jsx-indent': 1,
		'react/jsx-indent-props': 1,
		'react/jsx-key': 1,

		'react/jsx-max-depth': 0,

		'react/jsx-max-props-per-line': 1,
		'react/jsx-newline': 1,

		'react/jsx-no-bind': ['warn', { allowArrowFunctions: true }],

		'react/jsx-no-comment-textnodes': 1,
		'react/jsx-no-constructed-context-values': 1,
		'react/jsx-no-duplicate-props': 1,
		'react/jsx-no-leaked-render': 1,

		'react/jsx-no-literals': 0,

		'react/jsx-no-script-url': 1,
		'react/jsx-no-target-blank': 1,

		'react/jsx-no-useless-fragment': ['warn', { allowExpressions: true }],

		'react/jsx-one-expression-per-line': 1,
		'react/jsx-no-undef': 1,
		'react/jsx-curly-brace-presence': 1,
		'react/jsx-pascal-case': 1,
		'react/jsx-fragments': 1,
		'react/jsx-props-no-multi-spaces': 1,

		'react/jsx-props-no-spreading': 0, // spreading is useful

		'react/jsx-sort-default-props': 1,

		'react/jsx-sort-props': 0,

		'react/jsx-space-before-closing': 1,
		'react/jsx-tag-spacing': 1,
		'react/jsx-uses-react': 1,
		'react/jsx-uses-vars': 1,
		'react/jsx-wrap-multilines': 1,
		'react/no-invalid-html-attribute': 1,
		'react/no-access-state-in-setstate': 1,
		'react/no-adjacent-inline-elements': 1,
		'react/no-array-index-key': 1,
		'react/no-arrow-function-lifecycle': 1,
		'react/no-children-prop': 1,
		'react/no-danger': 1,
		'react/no-danger-with-children': 1,
		'react/no-deprecated': 1,
		'react/no-did-mount-set-state': 1,
		'react/no-did-update-set-state': 1,
		'react/no-direct-mutation-state': 1,
		'react/no-find-dom-node': 1,
		'react/no-is-mounted': 1,

		'react/no-multi-comp': 0, // can have several components per file

		'react/no-namespace': 1,
		'react/no-set-state': 1,
		'react/no-string-refs': 1,
		'react/no-redundant-should-component-update': 1,
		'react/no-render-return-value': 1,
		'react/no-this-in-sfc': 1,
		'react/no-typos': 1,
		'react/no-unescaped-entities': 1,
		'react/no-unknown-property': 1,
		'react/no-unsafe': 1,
		'react/no-unstable-nested-components': 1,
		'react/no-unused-class-component-methods': 1,

		'react/no-unused-prop-types': 0, // not using prop-types

		'react/no-unused-state': 1,
		'react/no-will-update-set-state': 1,
		'react/prefer-es6-class': 1,
		'react/prefer-exact-props': 1,
		'react/prefer-read-only-props': 1,
		'react/prefer-stateless-function': 1,

		'react/prop-types': 0, // not using prop-types

		'react/react-in-jsx-scope': 1,

		'react/require-default-props': 0, // not using prop-types

		'react/require-optimization': 0, // hmm
		'react/require-render-return': 1,
		'react/self-closing-comp': 1,
		'react/sort-comp': 1,

		'react/sort-prop-types': 0,

		'react/state-in-constructor': 1,
		'react/static-property-placement': 1,
		'react/style-prop-object': 1,
		'react/void-dom-elements-no-children': 1,
	},
} as const)
