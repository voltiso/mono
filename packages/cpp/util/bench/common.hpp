#pragma once

#include "v/allocator/Malloc"
#include "v/context"

// order may be important
inline auto g_malloc = v::allocator::Malloc();
// auto g_malloc = v::singleton::instance<v::allocator::Malloc>();
inline auto g_mallocGuard = v::context::Guard(g_malloc);
