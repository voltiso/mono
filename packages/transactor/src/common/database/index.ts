// â €â“¥ 2022     ğŸŒ©    ğŸŒ©     â €   â €
// â €         ğŸŒ© VÍ›oÍ›Í›Í›ltÍ›Í›Í›iÍ›Í›Í›Í›soÍ›Í›Í›.comâ €  â €â €â €

export * from './databaseUpdate'
export * from './fromDatabase'
export * from './isDocumentReference'
export * from './RefEntry'
export * from './toDatabase'
