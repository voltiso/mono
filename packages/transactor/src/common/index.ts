// â €â“¥ 2025     ğŸŒ©    ğŸŒ©     â €   â €
// â €         ğŸŒ© VÍ›oÍ›Í›Í›ltÍ›Í›Í›iÍ›Í›Í›Í›soÍ›Í›Í›.comâ €  â €â €â €

export * from './database'
export * from './pattern'
