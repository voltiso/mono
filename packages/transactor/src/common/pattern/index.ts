// â €â“¥ 2024     ğŸŒ©    ğŸŒ©     â €   â €
// â €         ğŸŒ© VÍ›oÍ›Í›Í›ltÍ›Í›Í›iÍ›Í›Í›Í›soÍ›Í›Í›.comâ €  â €â €â €

export * from './applyUnknownPathTokens'
export * from './getUnknownPathTokens'
export * from './PathMatches'
