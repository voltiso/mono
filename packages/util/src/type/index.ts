// â €â“¥ 2023     ðŸŒ©    ðŸŒ©     â €   â €
// â €         ðŸŒ© VÍ›oÍ›Í›Í›ltÍ›Í›Í›iÍ›Í›Í›Í›soÍ›Í›Í›.comâ €  â €â €â €

export * from './AlsoAccept'
export * from './brand'
export * from './compare'
export * from './define'
export * from './EmptyBraces'
export * from './If'
export * from './IntersectionFromUnion'
export * from './IsUnion'
export * from './number'
export * from './optional-argument'
export * from './Override'
export * from './parser'
export * from './Record'
export * from './Typeof'
export * from './TypeTag'
