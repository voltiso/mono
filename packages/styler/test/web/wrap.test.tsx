// â €â“¥ 2023     ðŸŒ©    ðŸŒ©     â €   â €
// â €         ðŸŒ© VÍ›oÍ›Í›Í›ltÍ›Í›Í›iÍ›Í›Í›Í›soÍ›Í›Í›.comâ €  â €â €â €

import { screen } from '@testing-library/react'
import type { Property } from 'csstype'
import type { ReactNode } from 'react'

import { style } from '~'

import { renderApp } from './common'

describe('wrap', () => {
	it('simple', () => {
		expect.hasAssertions()

		const Component = style('main').wrap(props => <div>{props.children}</div>)

		renderApp(<Component data-testid='test'>Hello</Component>)

		const myComponent = screen.getByTestId('test')

		expect(myComponent).toContainHTML(
			'<main data-testid="test"><div>Hello</div></main>',
		)
	})

	it('chain', () => {
		expect.hasAssertions()

		const Component = style('main')
			.wrap(props => <div>{props.children}</div>)
			.wrap(props => <button>{props.children}</button>)

		renderApp(<Component data-testid='test'>Hello</Component>)

		const myComponent = screen.getByTestId('test')

		expect(myComponent).toContainHTML(
			'<main data-testid="test"><div><button>Hello</button></div></main>',
		)
	})

	it('complex', () => {
		expect.hasAssertions()

		const H1 = style('h1')
		const MyFooter = style('footer').newCssProps({
			footerColor: (color: Property.Color) => ({ color }),
		})

		// @ts-expect-error no such element: `divA`
		;() => H1.wrap('divA')

		// good
		;() => H1.wrap('div')

		const MyFooter1 = MyFooter.newDomProp('data-testid', 'MyFooter1')

		const MyDiv = style('div')
			.newDomProp('data-testid', 'unset')
			.newProps({
				contentX: null as ReactNode,
				footerColor: 'red' as Property.Color,
			})
			.wrap(
				props => (
					<>
						<H1>Hello!</H1>

						<p>{props.contentX}</p>

						{props.children}
					</>
				),
				'span', // should not pass props
			)
			// wrap additional elements inside
			.wrap(
				MyFooter1, // should not pass props
				props => (
					<MyFooter footerColor={props.footerColor} data-testid='MyFooter2' />
				),
			)

		renderApp(
			<MyDiv
				data-testid='test'
				// eslint-disable-next-line react/forbid-component-props
				className='xxxxx'
				// eslint-disable-next-line react/forbid-component-props
				style={{ flex: 1 }}
				css={{ margin: 1 }}
				contentX='contentValue'
			>
				<div id='child' />
			</MyDiv>,
		)

		const myDiv = screen.getByTestId('test')

		expect(myDiv).toHaveClass('BDsg4w xxxxx', { exact: true })
		// expect(myDiv).toHaveClass('_0 xxxxx', { exact: true })
		expect(myDiv).toHaveStyle('flex: 1;')
		expect(myDiv).not.toHaveAttribute('css')
		expect(myDiv).not.toHaveAttribute('content')
		expect(myDiv).not.toHaveAttribute('footerColor')

		const myFooter2 = screen.getByTestId('MyFooter2')

		expect(myFooter2).toHaveStyle({ color: 'red' })

		const myFooter1 = screen.getByTestId('MyFooter1')

		expect(myFooter1).toHaveStyle({ color: 'unset' })

		// expect(myDiv).toContainHTML('aaaaa')
		expect(myDiv).toMatchSnapshot()
	})
})
