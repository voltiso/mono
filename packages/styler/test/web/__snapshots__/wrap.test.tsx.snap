// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`wrap complex 1`] = `
<div
  class="BDsg4w xxxxx"
  data-testid="test"
  style="flex: 1;"
>
  <h1>
    Hello!
  </h1>
  <p>
    contentValue
  </p>
  <footer
    data-testid="MyFooter1"
  />
  <footer
    class="rQJi6g"
    data-testid="MyFooter2"
  />
  <span />
</div>
`;
